### PR TITLE
Fix post-item alignment issues

### PR DESCRIPTION
**Pull request by jimrandomh**
There are a bunch of cases, which were varying degrees of subtly broken:

 * Comment expansion was totally broken when I made it fixed-height to address other issues. Now it's fixed height but not when expanded.
 * Vertical alignment of the title differed depending on whether there were any icons (eg curation star, AF omega) to the right of the title. This should now be consistent.
 * Title hover expansion created a weird white background when you hover with comments expanded. Now it only does that when not expanded.
 * Hovering (desktop layout) caused a slight shift in the vertical position of everything else on the line
 * The continue-reading sequence image has negative margins, which need to match the padding; adjusting the padding made it misaligned.

### CODE REVIEW

**Comment by Raemon** [comment]
Reviewing and testing. Stand by

**Comment by Raemon** [comment]
Looks good, and is at least not worse than status quo. Somewhat confused about why it was important to move to 10px instead of 12px (I think it's better on the margin to, um, keep the margins consistent. But not a hill I want to fight on hard enough to delay the PR merge)

File: packages/lesswrong/components/posts/PostsItem2.jsx [hunk 1]
```diff
@@ -32,10 +32,15 @@ const styles = (theme) => ({
...
+  fixedHeight: {
+    [theme.breakpoints.up('md')]: {
+      height: 48,
+    }
+  },
   postsItem: {
     display: "flex",
-    paddingTop: theme.spacing.unit*1.5,
-    paddingBottom: theme.spacing.unit*1.5,
+    paddingTop: 10,
```
Comment (Raemon) [inline packages/lesswrong/components/posts/PostsItem2.jsx]: why 10?

Comment (jimrandomh) [reply packages/lesswrong/components/posts/PostsItem2.jsx]: This affects the vertical position of elements other than the title (ie the karma, author, and comment-bubble), and they were a few pixels too low for some reason. This change moves them up. It does not affect the height of the PostsItem2 as a whole, which is fixed elsewhere.

**Pull request merged**